### PR TITLE
Try to work better on NFS

### PR DESCRIPTION
**Pull request by Hixie**
I haven't tested if this fixes NFS. I have tested that the locking still works on ext3.

cc @amirh for review

### CODE REVIEW

File: bin/flutter [hunk 1]
```diff
@@ -35,8 +35,9 @@ function path_uri() {
 
 function upgrade_flutter () {
   if hash flock 2>/dev/null; then
```
Comment (amirh) [inline bin/flutter]: I don't understand the trickery going on here.
Can you add comments explaining this?
What does `hash flock` do? Why are we locking fd 3?

Comment (Hixie) [reply bin/flutter]: Added some comments, PTAL. :-)

File: bin/flutter [hunk 2]
```diff
@@ -34,9 +34,41 @@ function path_uri() {
...
+  # shell built-in to determine if "flock" resolves to anything. If it
+  # does, we call it, otherwise we forget the whole thing.
+  #
+  # The second complication is NFS. On NFS, to obtain an exclusive
+  # lock you need a file descriptor that is open for writing, because
+  # NFS implements exclusive locks by writing, or some such. Thus, we
+  # ignore errors from flock. We do so by using the '|| true' trick,
+  # since we are running in a 'set -e' environment wherein all errors
+  # are fatal, and by redirecting all output to /dev/null, since
+  # users. will typically not care about errors from flock and are
```
Comment (amirh) [inline bin/flutter]: nit remove period

**Review by amirh — APPROVED** [review]
Thanks!
LGTM

File: bin/flutter (hunk 2 above)
Comment (Hixie) [reply bin/flutter]: good catch

**Comment by Hixie** [comment]
will land on green

**Pull request merged**